app/controllers/ContactHmrcController.scala

```diff
@@ -55,7 +55,7 @@ object ContactHmrcForm {
         .verifying("contact.comments.error.required", comment => comment.trim.nonEmpty)
         .verifying("contact.comments.error.length", comment => comment.length <= 2000),
       "isJavascript"     -> boolean,
-      "referrer"         -> text,
+      "referrer"         -> optional(text),
```

Comment:
I can't tell, why did this need to change to optional?

Reply:
> I can't tell, why did this need to change to optional?

I'll do my best to clarify using the FeedbackController as an example. It contains a renderFeedbackPage function that invokes the submit function. However, the submit function needs an Option[RedirectUrl]. To handle this, we first need to convert an Option[String] to Option[RedirectUrl]. But before converting, it's crucial to verify that the referer option does not contain the string "n/a." This check is necessary because "n/a" is not a valid redirect URL, and the conversion would fail. Using Option types in this context is beneficial primarily for these mapping operations. Let me know if that doesn't make sense and if there might be a better approach.

Reply:
aha I get you - if we had n/a though - wouldn't url.get(UnsafePermitAll) allow it through? maybe when we get the SafeRedirectUrl, instead we actually just want to get a String, and then in that method that takes a RedirectUrl => String we could have an early exit in the method like `if (url.unsafeValue.equalsIgnoreCase("n/a")) return url.unsafeValue`